search/includes/classes/class-search.php

```diff
@@ -1502,15 +1507,27 @@ public function filter__jetpack_widgets_to_include( $widgets ) {
 	}
 
 	/**
-	 * Filter for formatted_args in post queries
+	 * Filter for formatted_args in post queries (adjust the final query arguments before it gets passed to the request handler)
+	 *
+	 * @param array $formatted_args the prepared query arguments (payload)
+	 * @param array $args query vars
+	 * @param array $wp_query an instance of WP_Query
+	 * @return array
 	 */
-	public function filter__ep_post_formatted_args( $formatted_args ) {
+	public function filter__ep_post_formatted_args( $formatted_args, $args, $wp_query ) {
 		// Check if track_total_hits is set
 		// Don't override it if it is
 		if ( ! array_key_exists( 'track_total_hits', $formatted_args ) ) {
 			$formatted_args['track_total_hits'] = true;
 		}
 
+		// TODO: remove temporary rollout environment check
+		// Force the timeout for post search queries.
+		$is_prod = defined( 'VIP_GO_APP_ENVIRONMENT' ) && 'production' === VIP_GO_APP_ENVIRONMENT;
+		if ( ! isset( $formatted_args['timeout'] ) && apply_filters( 'vip_search_force_global_timeout', $is_prod ) ) {
```

Comment:
Are you intending to roll it out only for production? Or are you returning false somewhere with the filter?

Reply:
nice catch!